crates/bdk/tests/wallet.rs

```diff
@@ -526,6 +542,23 @@ fn test_create_tx_default_fee_rate() {
     assert_fee_rate!(psbt, fee.unwrap_or(0), FeeRate::default(), @add_signature);
 }
 
+// FIXME: Something is going on
+// thread 'test_legacy_create_tx_custom_fee_rate' panicked at 'assertion failed: `(left == right)`
+//  left: `FeeRate(9.537815)`,
+// right: `FeeRate(5.0)`'
+#[test]
+fn test_legacy_create_tx_custom_fee_rate() {
+    let (mut wallet, _) = get_funded_wallet(get_test_pkh());
+    let addr = wallet.get_address(New);
+    let mut builder = wallet.build_tx();
+    builder
+        .add_recipient(addr.script_pubkey(), 25_000)
+        .fee_rate(FeeRate::from_sat_per_vb(5.0));
+    let psbt = builder.finish().unwrap();
+
+    assert_eq!(psbt.fee_rate().unwrap(), FeeRate::from_sat_per_vb(5.0));
+}
+
```

Comment:
I need some help here. It is the only test that I honestly don't know what is going on...

Reply:
I am trying to debug this with:
```rust
#[test]
fn test_legacy_create_tx_custom_fee_rate() {
    let (mut wallet, _) = get_funded_wallet(get_test_pkh());
    let addr = wallet.get_address(New);
    let mut builder = wallet.build_tx();
    builder
        .add_recipient(addr.script_pubkey(), 25_000)
        .fee_rate(FeeRate::from_sat_per_vb(5.0));
    let mut psbt = builder.finish().unwrap();
    let psbt_before_sign = psbt.serialize_hex();
    println!("psbt before signining: {psbt_before_sign}");
    wallet.sign(&mut psbt, SignOptions::default()).unwrap();
    let psbt_after_sign = psbt.serialize_hex();
    println!("psbt after signining: {psbt_after_sign}");
    assert_eq!(psbt.fee_rate().unwrap(), FeeRate::from_sat_per_vb(5.0));
}
```

The `println!` of the hex-serialized PSBT before/after sign for PKH is:

```bash
psbt before signining: 70736274ff0100770100000001cd5114262b5ddbe15f003699dbe1032e3d44e74278c0542e41f50e4dc46ad3bd0000000000feffffff02395d0000000000001976a9148f77aaf68d832829d7c0015f6a42e647c1c29a5b88aca8610000000000001976a9148f77aaf68d832829d7c0015f6a42e647c1c29a5b88acd00700000001007401000000015b6de195f93238e277841e168c4153e285661a6a5b5601c3f0fa2f1acc12
711b0000000000ffffffff0250c30000000000001976a9148f77aaf68d832829d7c0015f6a42e647c1c29a5b88aca86100000000000016001488162cd490ad1686b11c96b421daa9e6ae08f62c0000000022060275d93539d503d824ad0c69a4c23ae480700489de09378ba32c1776cf86d6bb93048f77aaf60022020275d93539d503d824ad0c69a4c23ae480700489de09378ba32c1776cf86d6bb93048f77aaf60022020275d93539d503d824ad0c69a
4c23ae480700489de09378ba32c1776cf86d6bb93048f77aaf600

psbt after signining: 70736274ff0100770100000001cd5114262b5ddbe15f003699dbe1032e3d44e74278c0542e41f50e4dc46ad3bd0000000000feffffff02395d0000000000001976a9148f77aaf68d832829d7c0015f6a42e647c1c29a5b88aca8610000000000001976a9148f77aaf68d832829d7c0015f6a42e647c1c29a5b88acd00700000001007401000000015b6de195f93238e277841e168c4153e285661a6a5b5601c3f0fa2f1acc127
11b0000000000ffffffff0250c30000000000001976a9148f77aaf68d832829d7c0015f6a42e647c1c29a5b88aca86100000000000016001488162cd490ad1686b11c96b421daa9e6ae08f62c0000000022060275d93539d503d824ad0c69a4c23ae480700489de09378ba32c1776cf86d6bb93048f77aaf60022020275d93539d503d824ad0c69a4c23ae480700489de09378ba32c1776cf86d6bb93048f77aaf60022020275d93539d503d824ad0c69a4
c23ae480700489de09378ba32c1776cf86d6bb93048f77aaf600
```

They are the same!
However if I change the code above from `pkh(...)` to `wpkh(...)` I get different PSBT hexs:

```bash
psbt before signining: 70736274ff01007101000000012f7fcfa7ed07e83fb8a64d900d52d6313630d70b0304f018abcd0e8d370358770000000000feffffff02e75e000000000000160014ece526571053da86454c9e066779eaf9d1502e6aa861000000000000160014ece526571053da86454c9e066779eaf9d1502e6ad007000000010071010000000187e53680914906214fc892407033e379a27b743d5b0daa7ebdb996ce7f0d9e5d00000000
00ffffffff0250c3000000000000160014ece526571053da86454c9e066779eaf9d1502e6aa86100000000000016001488162cd490ad1686b11c96b421daa9e6ae08f62c0000000001011f50c3000000000000160014ece526571053da86454c9e066779eaf9d1502e6a2206032b0558078bec38694a84933d659303e2575dae7e91685911454115bfd64487e304ece52657002202032b0558078bec38694a84933d659303e2575dae7e91685911454115b
fd64487e304ece52657002202032b0558078bec38694a84933d659303e2575dae7e91685911454115bfd64487e304ece5265700

psbt after signining: 70736274ff01007101000000012f7fcfa7ed07e83fb8a64d900d52d6313630d70b0304f018abcd0e8d370358770000000000feffffff02e75e000000000000160014ece526571053da86454c9e066779eaf9d1502e6aa861000000000000160014ece526571053da86454c9e066779eaf9d1502e6ad007000000010071010000000187e53680914906214fc892407033e379a27b743d5b0daa7ebdb996ce7f0d9e5d000000000
0ffffffff0250c3000000000000160014ece526571053da86454c9e066779eaf9d1502e6aa86100000000000016001488162cd490ad1686b11c96b421daa9e6ae08f62c0000000001011f50c3000000000000160014ece526571053da86454c9e066779eaf9d1502e6a2206032b0558078bec38694a84933d659303e2575dae7e91685911454115bfd64487e304ece5265701070001086b0247304402204eded459c054780365d65e7f515d968d840c147f
60facad5a5eb9856e0860e1102204bea0ea30ce2e3202d6c407f7831611e1caa22ffa27751e467b2a6b6108601f10121032b0558078bec38694a84933d659303e2575dae7e91685911454115bfd64487e3002202032b0558078bec38694a84933d659303e2575dae7e91685911454115bfd64487e304ece52657002202032b0558078bec38694a84933d659303e2575dae7e91685911454115bfd64487e304ece5265700
```

Reply:
I am debugging this, I think I'll need to add a legacy descriptor version of `assert_fee_rate!`.

Reply:
Ok I've added a `assert_delta!` in b01a3864d952c5c257d239bfa331a5c3523103ee